### ISSUE TITLE
Artifacts in "The Saboteur" game.

### ISSUE THREAD

**Issue by rox4d** [iXit/Mesa-3D#356]
Wine d9d and dxvk are fine here, with nine on rx560 or hd7770 it looks like this screenshot.
![Tte_Sabotour](https://user-images.githubusercontent.com/34712490/111027552-00041280-8402-11eb-9722-fbf4b1b3310e.png)

Trace :
https://drive.google.com/file/d/1QM3g29m5Xs5R-cfQpIHNUrrXnGcvjke1/view?usp=sharing

**Comment by axeldavy** [comment]
Thank you, I can reproduce the issue

**Comment by rox4d** [comment]
a one more thing which was working with r600

**Comment by axeldavy** [comment]
Well, I could not figure out how to fix the issue.
Maybe it will be fixed one day by a random change in the driver.

**Comment by dungeon007** [comment]
On radeon driver flicker once then dissapear, meanwhile on amdgpu looks like cooking :D
Seems does not happen with AMD_DEBUG=nohyperz, kind of z-fighting from hz engine or maybe appear or get avoided by the luck...

**Comment by rox4d** [comment]
Thank for digging it, i can't see any difference between radeon and amdgpu drm on SI (R7 450 currently) here, but with AMD_DEBUG=nohyperz it looks correctly in both cases (it also helps with rx560).

**Comment by rox4d** [comment]
Lets try to inform radeonsi developers https://gitlab.freedesktop.org/mesa/mesa/-/issues/5690

**Comment by mirh** [comment]
Blip blop
I'm not a bot, but it seems like they have a fix incoming for this issue in the last link.

**Comment by lorn10** [comment]
Great news. This **radeonsi** related bug here will be resolved with Mesa MR [!22644](https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/22644). :smiley:

**Comment by rox4d** [comment]
Are you sure? i.e. your test it and it fixes issue?

**Comment by lorn10** [comment]
Yes, you can check the comment in Mesa bug 8716 [here](https://gitlab.freedesktop.org/mesa/mesa/-/issues/8716#note_1881174). 

As said, we will see it when Mesa MR [!22644](https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/22644) lands. :+1: 

And this here was in the end not a HyperZ problem (although disabling it helped). It was a problem of the **radeonsi** driver in conjunction with an old functionality called `VS_WINDOW_SPACE_POSITION`. That one seems to be missing in radeonsi and the emulation of that feature fixes the problem.

**Comment by rox4d** [comment]
> Yes,

? 
In pointed link i see a fixed branch https://gitlab.freedesktop.org/axeldavy/mesa/-/commits/unsupported_features_for_merge
(with env var force_features_emulation=true) but pointed MR are not about it.

**Comment by lorn10** [comment]
When you build Mesa with MR [!22644](https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/22644) then you can verify oneself if this radeonsi bug here is resolved or not. :+1:

**Comment by rox4d** [comment]
So why you does`t do it before writing?

**Comment by axeldavy** [comment]
It is not fixed by the MR.

**Comment by lorn10** [comment]
> So why you does`t do it before writing?

I have currently no radeonsi hardware, only r600, crocus and old old r300 stuff.

> It is not fixed by the MR.

:disappointed: So the [!22644](https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/22644) is only one piece. Whatever, sorry for the noise.
